src/main/java/nl/knaw/dans/easy/sword2examples/Common.java

```diff
@@ -194,21 +194,30 @@ public static CloseableHttpResponse sendChunk(DigestInputStream dis, int size, S
         byte[] chunk = readChunk(dis, size);
         String md5 = new String(Hex.encodeHex(dis.getMessageDigest().digest()));
         HttpUriRequest request = RequestBuilder.create(method).setUri(uri).setConfig(RequestConfig.custom()
-        /*
-         * When using an HTTPS-connection EXPECT-CONTINUE must be enabled, otherwise buffer overflow may follow
-         */
-        .setExpectContinueEnabled(true).build()) //
+            /*
+             * When using an HTTPS-connection EXPECT-CONTINUE must be enabled, otherwise buffer overflow may follow
+             */
+            .setExpectContinueEnabled(true).build()) //
                 .addHeader("Content-Disposition", String.format("attachment; filename=%s", filename)) //
                 .addHeader("Content-MD5", md5) //
                 .addHeader("Packaging", BAGIT_URI) //
                 .addHeader("In-Progress", Boolean.toString(inProgress)) //
                 .setEntity(new ByteArrayEntity(chunk, ContentType.create(mimeType))) //
                 .build();
+        addXAuthorizationToRequest(request);
         CloseableHttpResponse response = http.execute(request);
         // System.out.println("Response received.");
         return response;
     }
 
+    private static HttpUriRequest addXAuthorizationToRequest(HttpUriRequest request) throws Exception {
+        File autValueFile = new File("x-auth-value.txt");
+        if (autValueFile.exists()) {
+            request.addHeader("X-Authorization", FileUtils.readFileToString(autValueFile).trim());
```

Comment:
Usually files end on a newline (unix convention); does this remove that newline? I know `.trim()` removes whitespaces at the start/end of the string, but I'm not sure it also removes that newline.

Reply:
The newline character is whitespace, AFAIK.

Reply:
More info at: https://stackoverflow.com/a/18169122

Reply:
Great! I assume you tested this as well 😜